src/build/env.py

```diff
@@ -1,241 +1,215 @@
 
-    @classmethod
-    def for_current(cls):  # type: () -> IsolatedEnvironment
-        """
-        Creates an isolated environment for the current interpreter
+    def install(self, requirements):  # type: (Iterable[str]) -> None
         """
-        remove_paths = os.environ.get('PYTHONPATH', '').split(os.pathsep)
-        # also remove the pth file adding this project (when installed via --develop install)
-        for path in sys.path:
-            if os.path.exists(os.path.join(path, 'build.egg-info')):
-                remove_paths.append(path)  # pragma: no cover
+        Installs the specified PEP 508 requirements on the environment
 
-        return cls(remove_paths)
+        :param requirements: PEP-508 requirement specification to install
 
-    def _replace_env(self, key, new):  # type: (str, Optional[str]) -> None
-        """
-        Replaces an environment variable
+        :note: Passing non PEP 508 strings will result in undefined behavior, you *should not* rely on it. It is \
+               merely an implementation detail, it may change any time without warning.
         """
-        if not new:  # pragma: no cover
+        if not requirements:
             return
 
-        self._env[key] = os.environ.get(key, None)
-        os.environ[key] = new
+        with tempfile.NamedTemporaryFile('w+', prefix='build-reqs-', suffix='.txt', delete=False) as req_file:
+            req_file.write(os.linesep.join(requirements))
+        try:
+            cmd = [
+                self._pip_executable,
+                # on python2 if isolation is achieved via environment variables, we need to ignore those while calling
+                # host python (otherwise pip would not be available within it)
+                '-{}m'.format('E' if self._pip_executable == self.executable and sys.version_info[0] == 2 else ''),
+                'pip',
+                'install',
+                '--prefix',
+                self.path,
+                '--ignore-installed',
+                '--no-warn-script-location',
+                '-r',
+                os.path.abspath(req_file.name),
+            ]
+            subprocess.check_call(cmd)
+        finally:
+            os.unlink(req_file.name)
 
-    def _pop_env(self, key):  # type: (str) -> None
-        """
-        Removes an environment variable
-        """
-        self._env[key] = os.environ.pop(key, None)
 
-    def _restore_env(self):  # type: () -> None
-        """
-        Restores the initial environment variables
-        """
-        for key, val in self._env.items():
-            if val is None:
-                os.environ.pop(key, None)
-            else:
-                os.environ[key] = val
+if sys.version_info[0] == 2:  # noqa: C901 # disable if too complex
 
-    def _get_env_path(self, path):  # type: (str) -> Optional[str]
-        """
-        Returns sysconfig path from our environment
+    def _create_isolated_env(path):  # type: (str) -> Tuple[str, str]
```

Comment:
Wouldn't make sense for this to be in `IsolatedEnvBuilder`?

Reply:
Well, maybe, however, these are implementation detail methods. If we move them, you'd need to make these static methods and increase the indent by one level. So overall, probably not worth the change.